src/commands/templates.rs

```diff
@@ -468,15 +475,23 @@ impl List {
     }
 
     fn print_templates_json(&self, list_results: &ListResults) -> anyhow::Result<()> {
-        let json_vals: Vec<_> = list_results
-            .templates
-            .iter()
+        let json_vals: Vec<_> = self
+            .filter_templates_with_tags(&list_results.templates)
+            .into_iter()
             .map(json_list_format)
             .collect();
+
         let json_text = serde_json::to_string_pretty(&json_vals)?;
         println!("{}", json_text);
         Ok(())
     }
+
+    fn filter_templates_with_tags<'i>(&'i self, templates: &'i [Template]) -> Vec<&'i Template> {
+        templates
+            .iter()
+            .filter(|tpl| self.tags.iter().all(|tag| tpl.tags().contains(tag)))
```

Comment:
I found the nested expression with 4 occurrences of the word `tag` in it a bit hard to follow.  I wonder if rather than returning the `tags` collection, `Template` should have a `has_all_tags` or `matches_all_tags` method so this could be `filter(|tpl| tpl.has_all_tags(&self.tags))`?

Reply:
I was hoping this would you'd say that. Will factor out some helpers to prettify this combinator.